modules/wallet/encrypt.go

```diff
@@ -193,29 +193,48 @@ func (w *Wallet) managedUnlock(masterKey crypto.TwofishKey) error {
 	subscribed := w.subscribed
 	w.mu.RUnlock()
 	if !subscribed {
-		// Subscription can take a while, so spawn a goroutine to print the
-		// wallet height every few seconds. (If subscription completes
-		// quickly, nothing will be printed.)
-		done := make(chan struct{})
-		go w.rescanMessage(done)
-		defer close(done)
-
-		err = w.cs.ConsensusSetSubscribe(w, lastChange)
-		if err == modules.ErrInvalidConsensusChangeID {
-			// something went wrong; resubscribe from the beginning
-			err = dbPutConsensusChangeID(w.dbTx, modules.ConsensusChangeBeginning)
+		if primarySeedProgress == 0 {
+			// If the wallet has not been used, we don't need to scan the entire
+			// blockchain looking for relevant transactions. Instead, just grab
+			// the current height and subscribe to future blocks.
+			err = w.cs.ConsensusSetSubscribe(w, modules.ConsensusChangeRecent)
 			if err != nil {
-				return fmt.Errorf("failed to reset db during rescan: %v", err)
+				return fmt.Errorf("wallet subscription failed: %v", err)
 			}
-			err = dbPutConsensusHeight(w.dbTx, 0)
+			w.mu.Lock()
+			err = dbPutConsensusHeight(w.dbTx, w.cs.Height())
+			w.mu.Unlock()
```

Comment:
In theory the cc could provide the input values, since it knows the value of every unspent output, right? I assume that would require modifying the consensus persist structures though. Same with the height.

I think it's worth contacting the exchanges to see if this will actually break anything. Given that Bitcoin does not provide the value of every input, I would be surprised if the exchanges required it. I don't want to tell users that Sia is eating up gigabytes of RAM+disk and wasting hours of their time to store values that they will never see or care about because we _think_ that someone might need them.